mlir/lib/Dialect/Utils/IndexingUtils.cpp

```diff
@@ -8,6 +8,8 @@
 
 #include "mlir/Dialect/Utils/IndexingUtils.h"
 
+#include "mlir/Dialect/NVGPU/IR/NVGPUDialect.h"
+#include "mlir/Dialect/NVGPU/Transforms/Utils.h"
```

Comment:
I don't think `IndexingUtils.cpp` can take a dependency on NVGPU or Vector. I assume the portion of the helper that you want to reuse isn't specific to any NVGPU ops; maybe split out the vector and memref portions and make them available in the vector dialect, or just duplicate the portion of the helper that is needed in AMDGPU?

Reply:
That is a good point. We can discuss about it today but probably I will create separated .cpp util file for that. Main reason for this patch to avoid code duplication as mush as we can.